tools/src/main/scala/caliban/tools/Codegen.scala

```diff
@@ -8,16 +8,17 @@ import zio.{ Task, UIO }
 object Codegen {
   def generate(
     arguments: Options,
-    writer: (Document, String, Option[String], String) => String
+    writer: (Document, String, Option[String], String, Map[String, String]) => String
   ): Task[Unit] = {
-    val s           = ".*/scala/(.*)/(.*).scala".r.findFirstMatchIn(arguments.toPath)
-    val packageName = arguments.packageName.orElse(s.map(_.group(1).split("/").mkString(".")))
-    val objectName  = s.map(_.group(2)).getOrElse("Client")
-    val effect      = arguments.effect.getOrElse("zio.UIO")
-    val loader      = getSchemaLoader(arguments.schemaPath, arguments.headers)
+    val s            = ".*/scala/(.*)/(.*).scala".r.findFirstMatchIn(arguments.toPath)
```

Comment:
Play apps can actually use `app` at the top-level with no subtree. I wonder if we should just add that here:

```scala
.*/(?:scala|app)/*(.*)/(.*).scala
```

wdyt @ghostdogpr

Reply:
Sure!